packages/aws-cdk-lib/aws-ecs/lib/ec2/ec2-service.ts

```diff
@@ -249,7 +249,7 @@ export class Ec2Service extends BaseService implements IEc2Service {
    */
   private validateEc2Service(): string[] {
     const ret = new Array<string>();
-    if (!this.cluster.hasEc2Capacity) {
+    if (!this.daemon && !this.cluster.hasEc2Capacity) {
```

Comment:
I'm not sure I see how this is any different from the closed issue that is related. The ask is to be able to add services to a cluster that don't have any capacity. Sounds like you are proposing that you found a loophole in the original issue because we never specifically mentioned `daemon` services 😁

I think what we need instead is a new way of managing capacity in clusters. You need capacity in a cluster so we don't want to remove that check, but we should have a way to tell the cluster that you have capacity.

Reply:
@corymhall I completely understand your point of view. As I said in https://github.com/aws/aws-cdk/issues/25306#issuecomment-1524057191 - I don't necessarily believe we should fail validation for lack of capacity. I purposely kept this PR small in scope because I think it's more easily defensible, whereas the larger question still needs to be resolved.

Reply:
@otterley I disagree with your comment. If we have reason to believe that a users configuration will fail once deployed we should fail synthesis.

The bottom line is that multiple CDK team members have said the type of change that we will support. This PR falls under the same use case.